src/utils/arguments.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Args for remote function call.
+ *
+ * This class can serialize javascript native types into bytes, in order to
+ * make smart-contract function call easier.
+ *
+ */
+export default class Args {
+  private offset: number = 0;
+  private serialized: string = "";
+
+  /**
+   *
+   * @param {string} serialized
+   */
+  constructor(serialized: string = "") {
+    this.serialized = serialized;
+  }
+
+  /**
+   * Returns the serialized string to pass to CallSC.
+   *
+   * @return {string} the serialized string
+   */
+  serialize(): string {
+    return this.serialized;
+  }
+
+  // Getters
+
+  /**
+   * Returns the deserialized string.
+   *
+   * @return {string} the string
+   */
+  nextString(): string {
+    const length = Number(this.nextU32());
+    const end = this.offset + length;
+    const result = this.serialized.slice(this.offset, end);
+    this.offset = end;
+    return result;
+  }
+
+  /**
+   * Returns the deserialized number.
+   *
+   * @return {BigInt}
+   */
+  nextU32(): BigInt {
+    const byteArray = this.fromByteString(this.serialized);
+
+    if (byteArray.length - this.offset < 4) {
+      return BigInt(NaN);
+    }
+
+    let value = BigInt(0);
+
+    // encoding is little endian
+    for (let i = 3; i >= 1; --i) {
+      value = (value | BigInt(byteArray[this.offset + i])) << BigInt(8);
+    }
+
+    value = value | BigInt(byteArray[this.offset]);
+    this.offset += 4;
```

Comment:
You should remove the magic value here. My proposal :

```suggestion
   numBytes = sizeof<u32>();
    if (byteArray.length - this.offset < numBytes) {
      return BigInt(NaN);
    }

    let value = BigInt(0);

    // encoding is little endian
    for (let i = NumBytes-1; i > 0; --i) {
      value = (value | BigInt(byteArray[this.offset + i])) << BigInt(8);
    }

    value = value | BigInt(byteArray[this.offset]);
    this.offset += numBytes;
```

Reply:
there is no sizeof in typescript